Sources/NetworkService/InternalError/NetworkServiceInternalError.swift

```diff
@@ -1,6 +1,5 @@
-import Foundation
 /// Defines a set of internal errors that can occur during using the `NetworkService`.
-public enum InternalError: Error {
+public enum NetworkServiceInternalError: Error, CaseIterable {
```

Comment:
The `LocalizedError` inherits from `Error`. Why do we need to conform to `Error`?